Geofencing/app/src/main/java/com/google/android/gms/location/sample/geofencing/MainActivity.java

```diff
@@ -26,406 +26,411 @@
+  private GeofencingClient mGeofencingClient;
 
-        // Get the UI widgets.
-        mAddGeofencesButton = findViewById(R.id.add_geofences_button);
-        mRemoveGeofencesButton = findViewById(R.id.remove_geofences_button);
+  /** The list of geofences used in this sample. */
+  private ArrayList<Geofence> mGeofenceList;
 
-        // Empty list for storing geofences.
-        mGeofenceList = new ArrayList<>();
+  /** Used when requesting to add or remove geofences. */
+  private PendingIntent mGeofencePendingIntent;
 
-        // Initially set the PendingIntent used in addGeofences() and removeGeofences() to null.
-        mGeofencePendingIntent = null;
+  // Buttons for kicking off the process of adding or removing geofences.
+  private Button mAddGeofencesButton;
+  private Button mRemoveGeofencesButton;
 
-        setButtonsEnabledState();
+  private PendingGeofenceTask mPendingGeofenceTask = PendingGeofenceTask.NONE;
 
-        // Get the geofences used. Geofence data is hard coded in this sample.
-        populateGeofenceList();
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.main_activity);
 
-       mGeofencingClient = LocationServices.getGeofencingClient(this);
-    }
+    // Get the UI widgets.
+    mAddGeofencesButton = findViewById(R.id.add_geofences_button);
+    mRemoveGeofencesButton = findViewById(R.id.remove_geofences_button);
 
-    @Override
-    public void onStart() {
-        super.onStart();
+    // Empty list for storing geofences.
+    mGeofenceList = new ArrayList<>();
 
-        if (!checkPermissions()) {
-            requestPermissions();
-        } else {
-            performPendingGeofenceTask();
-        }
-    }
+    // Initially set the PendingIntent used in addGeofences() and removeGeofences() to null.
+    mGeofencePendingIntent = null;
 
-    /**
-     * Builds and returns a GeofencingRequest. Specifies the list of geofences to be monitored.
-     * Also specifies how the geofence notifications are initially triggered.
-     */
-    private GeofencingRequest getGeofencingRequest() {
-        GeofencingRequest.Builder builder = new GeofencingRequest.Builder();
+    setButtonsEnabledState();
 
-        // The INITIAL_TRIGGER_ENTER flag indicates that geofencing service should trigger a
-        // GEOFENCE_TRANSITION_ENTER notification when the geofence is added and if the device
-        // is already inside that geofence.
-        builder.setInitialTrigger(GeofencingRequest.INITIAL_TRIGGER_ENTER);
+    // Get the geofences used. Geofence data is hard coded in this sample.
+    populateGeofenceList();
 
-        // Add the geofences to be monitored by geofencing service.
-        builder.addGeofences(mGeofenceList);
+    mGeofencingClient = LocationServices.getGeofencingClient(this);
+  }
 
-        // Return a GeofencingRequest.
-        return builder.build();
-    }
+  @Override
```

Comment:
This goes against the best practices. I understand for the sample might be okay, but I would consider moving this logic under a button/CTA in the screen.